agent/consul/client.go

```diff
@@ -186,7 +186,7 @@ func NewClientLogger(config *Config, logger *log.Logger, tlsConfigurator *tlsuti
 	go c.routers.Start()
 
 	// Start GRPC client.
-	c.grpcClient = NewGRPCClient(logger, c.routers)
+	c.grpcClient = NewGRPCClient(logger, c.routers, tlsConfigurator.OutgoingRPCWrapper())
```

Comment:
@i0rek can you sanity check this for us please? Will this work for TLS cert reload for example if we only pass this in on init?

Reply:
Correct @banks. This won't work with reloading. @kyhavlov if possible you should pass in the `tlsConfigurator` and call the wrapper function as late as possible.